gap/stack.gd

```diff
@@ -0,0 +1,51 @@
+##
+#Y  Copyright (C) 2017 The GAP Group
+##
+
+#! @Chapter Stacks
+#!
+#! A stack is a deque where items can be <C>Push</C>ed onto the stack, and
+#! the top item can be <C>Pop</C>ped off the stack.
+#!
+#! Stacks are wrapped GAP Plain lists.
+
+
+#! @Section API
+#!
+#! @Description
+#! Category of heaps
+DeclareCategory("IsStack", IsCollection);
+BindGlobal( "StackFamily", NewFamily("StackFamily") );
+
+#TODO Do we want to use constructors?
+# Arguments are a heap filter, a comparison function, and maybe some data?
+DeclareConstructor("NewStack", [IsStack]);
```

Comment:
As long as there is only one implementation: no. This is only useful (if at all) when there are multiple implementations